src/session/session_dhandle.c

```diff
@@ -509,6 +511,23 @@ __wt_session_get_btree_ckpt(WT_SESSION_IMPL *session, const char *uri, const cha
     do {
         ret = 0;
 
+        /*
+         * Save the checkpoint generation number and the checkpoint's state to detect races.
+         * Note that we must save the generation number before the checkpoint's state. Indeed, if we
+         * save the generation number after, we could have the following scenario:
+         *
+         * 1) The checkpoint's state is evaluated to false.
+         * 2) -- Race, checkpoint starts.
+         * 3) The checkpoint generation number is saved and is equal to the latest one, we cannot
+         * detect the race.
+         *
+         * By saving the generation number before, if there is a race, the saved generation number
+         * will not be equal to the latest one. Since we want both variables to be read in as early
```

Comment:
Nit: `Since we want` -> `We want`

Reply:
Fixed in 1d48cc2f1ab363cece4ae1063c8a24c22540958d